emoji-ui/src/components/Pagination.js

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import PaginationActions from '../store/actions/Pagination';
+import { totalPageNumberSelector } from '../store/selectors/Pagination';
+import './Pagination.css';
+
+class PaginationBase extends Component {
+  static propTypes = {
+    currentPage: PropTypes.number.isRequired,
+    totalPageNumber: PropTypes.number.isRequired,
+    paginationActions: PropTypes.shape({
+      setCurrentPage: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  constructor() {
+    super(...arguments);
+    this._onInputChange = this._onInputChange.bind(this);
+  }
+
+  _onInputChange(e) {
+    const { setCurrentPage } = this.props.paginationActions;
+    setCurrentPage(e.target.value);
```

Comment:
suggestion: 

```
const pageNum = parseInt(e.target.value);
if (!sNaN(pageNum)) {
  setCurrentPage(pageNum);
}
```

Reply:
will do!

Reply:
i did `setCurrentPage(isNaN(pageNum) ? 0 : pageNum);` otherwise i can not remove the entire number to set another number.